### PR TITLE
Dash Kim

### PR DESCRIPTION
**Pull request by dashkim**
Project-3

### CODE REVIEW

File: vocab/templates/vocab.html [hunk 1]
```diff
@@ -73,32 +73,80 @@ <h1>Words</h1>
...
+                // Prevent normal form submission
+                $("#entry").submit(function(event) {
+                    event.preventDefault();
+                });
+        
+                // Cache DOM elements
+                var $attemptInput = $("#attempt");
+                var $foundText = $("#foundText");
+                var $results = $("#results");
+                var $messages = $("#messages");
```
Comment (afloresN9ne) [inline vocab/templates/vocab.html:101]: This approach works perfectly fine, but from my own perspective creating variables for the $("#  ") formats in this instance doesn't really provide any improvements to the original format. Maybe for next time you could shorten the variable names a little more so that you benefit more from using the variables then the original format.

File: vocab/flask_vocab.py [hunk 2]
```diff
@@ -79,48 +80,51 @@ def success():
...
-    elif not matched:
-        flask.flash("{} isn't in the list of words".format(text))
-    elif not in_jumble:
-        flask.flash(
-            '"{}" can\'t be made from the letters {}'.format(text, jumble))
+    if matched and in_jumble:
+        if text not in matches:
+            # Cool, they found a new word
+            matches.append(text)
+            flask.session["matches"] = matches
```
Comment (afloresN9ne) [inline vocab/flask_vocab.py:103]: personally i like this representation of the code better then the original. I feel like this way is easier to understand so good job.

**Comment by afloresN9ne** [comment]
Overall everything I saw was great besides one small change I wouldn't personally have done, but either way it still works. I would say for the future maybe add a few more comments into your code just so that things can be a little clearer but besides that everything looks good to me.

File: vocab/flask_vocab.py [hunk 3]
```diff
@@ -79,48 +80,51 @@ def success():
...
+        else:
+            messages.append("You already found {}".format(text))
     else:
-        app.logger.debug("This case shouldn't happen!")
-        assert False  # Raises AssertionError
+        if not matched:
+            messages.append("{} isn't in the list of words".format(text))
+        if not in_jumble:
+            messages.append(
+                '"{}" can\'t be made from the letters {}'.format(text, jumble))
```
Comment (djmurphy6) [inline vocab/flask_vocab.py:111]: I think that you could benefit from a few more comments because this might be hard to follow for someone not familiar with the project

File: vocab/flask_vocab.py [hunk 4]
```diff
@@ -79,48 +80,51 @@ def success():
...
+    # Check if the player has succeeded
+    success = len(matches) >= flask.session["target_count"]
+
+    # Log the current list of matches
+    app.logger.debug("matches: ")
+    for match in matches:
+        app.logger.debug(match)
+
+    # Prepare the JSON response
+    result = {"matches": matches, "messages": messages, "success": success}
```
Comment (djmurphy6) [inline vocab/flask_vocab.py:122]: I like how you packaged all of these into "result" it makes it very clean and I didn't know you could do that

**Review by djmurphy6 — COMMENTED** [review]
Dash, I found your code very polished but there were quite a few times where it would have been nice to have more comments. It seems like you are familiar and comfortable with flask and javascript and that showed in your concise code that from my perspective was working efficiently without using workarounds. When comparing to my project I was able to see that you had a more direct way of passing information from your flask to your javascript that made sense and was relatively easy to follow. The only thing I noticed was the entry field not auto clearing when you got a word correct but that wasn't explicitly stated in the README so I'm not sure if it's necessary.

While your code is easy to follow for someone already familiar with the project, it would be quite difficult to jump in and try to understand the logic. I think you could really benefit from more comments to help the person reading your code flow through it rather than trying to piece things together. That being said, your code is overall very clean without any distractions or workaround taking away from my understanding of the meat / functionality of your code. Great job!

File: vocab/templates/vocab.html [hunk 5]
```diff
@@ -73,32 +73,80 @@ <h1>Words</h1>
...
+                        var matches = data.result.matches;
+                        var messages = data.result.messages;
+                        var success = data.result.success;
+        
+                        updateResults(matches, messages);
+                        words = matches;
+        
+                        if (success) {
+                            window.location.replace("/success");
+                        }
```
Comment (djmurphy6) [inline vocab/templates/vocab.html:143]: The entry field doesn't auto-clear when I get a word right
